sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_client.hpp

```diff
@@ -202,6 +202,23 @@ namespace Azure { namespace Storage { namespace Blobs {
         const SetBlobAccessTierOptions& options = SetBlobAccessTierOptions(),
         const Azure::Core::Context& context = Azure::Core::Context()) const;
 
+    /**
+     * @brief Copies data at from the source to this blob synchronously.
+     *
+     * @param sourceUri Specifies the URL of the source blob. The value may be a URL of up to 2 KB
+     * in length that specifies a blob. The value should be URL-encoded as it would appear in a
+     * request URI. The source blob must either be public or must be authorized via a shared access
+     * signature. If the size of thesource blob is greater than 256 MB, the request will fail with
+     * 409 (Conflict). The blob type of the source blob has to be block blob.
+     * @param options Optional parameters to execute this function.
+     * @param context Context for cancelling long running operations.
+     * @return A CopyBlobFromUriResult describing the copy result.
+     */
+    Azure::Response<Models::CopyBlobFromUriResult> CopyFromUri(
```

Comment:
Is the reason why this isn't a long-running oepration

The LRO version, `StartCopyFromUri()`, returns `BlobProperties`, why is this returning `CopyBlobFromUriResult`?

https://github.com/Azure/azure-sdk-for-cpp/blob/fda88bf7b8389575b9d973821efc604442e10493/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp#L171

Is that not also synchronous? When does the user call one vs the other?

Reply:
`StartCopyFromUri` starts the copy operation and then returns before the operation is completed. Storage service will continue copying the blob in the background. Client side can check if the copy is completed by getting the blobs' properties, which includes a field of copy status. That's why this is an LRO operation and why `Poll()` returns `BlobProperties`.

In constrast, `CopyFromUri` copies the blob synchronously, which means by the time the REST API returns, the copy operation is either finished or failed. That's why it's not an LRO. The return type `CopyBlobFromUriResult` is just to follow our naming pattern, `[Operation]Result`.

Reply:
`CopyFromUri` can only be used to copy small blobs (<256MB, IIRC). `StartCopyFromUri` can be used on any size of blobs.

> When does the user call one vs the other?

If a customer knows for sure a blob is relatively small and wants to write simple code, he can just write one line of code
`blob_client.CopyFromUri(url);`, which makes exactly one REST API call.
If he knows the blob can be large and the copy operation can take quite some time to finish, he'll call `StartCopyFromUri`.

Reply:
Ok, gotcha, that makes sense now. Thanks for clarifying.